### PR TITLE
bpo-29849: fix memory leak in import_from

### PR DESCRIPTION
**Pull request by zhangyangyu**

### CODE REVIEW

**Review by serhiy-storchaka — APPROVED** [review]

**Comment by brettcannon** [comment]
The Travis failures were probably due to GH-697 which has now been fixed in GH-718. Please rebase your branch to pick up the change.

File: Python/ceval.c [hunk 1]
```diff
@@ -5056,17 +5057,21 @@ import_from(PyObject *v, PyObject *name)
...
         PyErr_Clear();
-        PyErr_SetImportError(
-            PyUnicode_FromFormat("cannot import name %R from %R (unknown location)",
-                name, pkgname_or_unknown),
-            pkgname, NULL);
+        errmsg = PyUnicode_FromFormat(
+            "cannot import name %R from %R (unknown location)",
+            name, pkgname_or_unknown
+        );
+        /* doesn't check NULL for errmsg, PyErr_SetImportError will catch */
```
Comment (brettcannon) [inline Python/ceval.c]: I don't think the comment is necessary long-term.

**Review by brettcannon — CHANGES_REQUESTED** [review]
Minor doc removal, otherwise LGTM.

File: Python/ceval.c (hunk 1 above)
Comment (zhangyangyu) [reply Python/ceval.c]: This comment is to make others know the error check is left out deliberately. It is not an oversight.
Do you think it's a must to remove it?

File: Python/ceval.c [hunk 2]
```diff
@@ -5063,17 +5064,21 @@ import_from(PyObject *v, PyObject *name)
...
+        PyErr_SetImportError(errmsg, pkgname, NULL);
     } else {
```
Comment (brettcannon) [inline Python/ceval.c]: While you're here you might as well fix the formatting for this.

File: Python/ceval.c (hunk 1 above)
Comment (brettcannon) [reply Python/ceval.c]: If you want to keep it then please change it to "/* NULL check for errmsg done by PyErr_SetImportError. */".

File: Python/ceval.c [hunk 3]
```diff
@@ -5063,17 +5064,21 @@ import_from(PyObject *v, PyObject *name)
...
-        PyErr_SetImportError(
-            PyUnicode_FromFormat("cannot import name %R from %R (%S)",
-                name, pkgname_or_unknown, pkgpath),
-            pkgname, pkgpath);
+        errmsg = PyUnicode_FromFormat(
+            "cannot import name %R from %R (%S)",
+            name, pkgname_or_unknown, pkgpath
+        );
+        PyErr_SetImportError(errmsg, pkgname, pkgpath);
```
Comment (brettcannon) [inline Python/ceval.c]: Need a similar comment here if you are going to leave it in.

**Review by brettcannon — CHANGES_REQUESTED** [review]

**Comment by zhangyangyu** [comment]
Thanks @serhiy-storchaka and @brettcannon !

**Pull request merged**